etna/records/tests/test_views.py

```diff
@@ -722,3 +725,100 @@ def test_invalid_response_from_kong_raises_404(self):
         response = self.client.get("/records/images/C123456/11000000000000000000/")
 
         self.assertEquals(response.status_code, 404)
+
+
+class RecordDetailBackToSearchTest(TestCase):
+    def setUp(self):
+        responses.add(
+            responses.GET,
+            "https://kong.test/data/fetch",
+            json=create_response(
+                records=[
+                    create_record(iaid="C13359805"),
+                ]
+            ),
+        )
+
+        self.record_detail_url = reverse(
+            "details-page-machine-readable", kwargs={"iaid": "C13359805"}
+        )
+
+        self.expected_button_link_gen_value_fmt = '<a class="cta-primary-panel__link" href="{back_to_search_url}" data-link-type="Link" data-link="Back to search results" data-component-name="Navigation">\n\n\n\n<svg\n    class="icon icon--chevron-left cta-primary-panel__link-icon"\n    aria-hidden="true"\n    focusable="false"\n>\n    <use xlink:href="#chevron-left" href="#chevron-left"></use>\n</svg>\nBack to search results</a>'
+        self.back_to_search_url_timestamp = timezone.now()
+
+    @responses.activate
+    def test_back_to_search_render_with_catalogue_search_within_expiry(self):
+        """navigation to record details from previous search (session is set since its coming from search catalogue)"""
+
+        # raw search value: "/search/catalogue/?sort_by=title&q=london&filter_keyword=paper&level=Item&collection=ADM&collection=BT&closure=Open+Document%2C+Open+Description&opening_start_date_0=&opening_start_date_1=&opening_start_date_2=1900&opening_end_date_0=&opening_end_date_1=&opening_end_date_2=2020&per_page=20&sort_order=asc&display=list&page=2&group=tna"
+        search_url_gen_html_resp = "%2Fsearch%2Fcatalogue%2F%3Fsort_by%3Dtitle%26q%3Dlondon%26filter_keyword%3Dpaper%26level%3DItem%26collection%3DADM%26collection%3DBT%26closure%3DOpen%2BDocument%252C%2BOpen%2BDescription%26opening_start_date_0%3D%26opening_start_date_1%3D%26opening_start_date_2%3D1900%26opening_end_date_0%3D%26opening_end_date_1%3D%26opening_end_date_2%3D2020%26per_page%3D20%26sort_order%3Dasc%26display%3Dlist%26page%3D2%26group%3Dtna"
```

Comment:
I'm not sure about the variable name here... feels like the original one is a better fit

Reply:
I kept this `search_url_gen_html_resp`  separate so that we focus only on the url here, the other bit - `expected_button_link_gen_value_fmt` may change independently - ex. the class name or so.

Reply:
A small change updated - removed raw search value comment.